cachito/workers/tasks/npm.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import base64
+import logging
+
+from cachito.errors import CachitoError
+from cachito.workers import nexus
+from cachito.workers.config import validate_nexus_config
+from cachito.workers.paths import RequestBundleDir
+from cachito.workers.pkg_managers.general import (
+    update_request_with_config_files,
+    update_request_with_deps,
+    update_request_with_packages,
+)
+from cachito.workers.pkg_managers.general_js import (
+    generate_npmrc,
+    get_js_proxy_repo_name,
+    get_js_proxy_username,
+    get_npm_token,
+    finalize_nexus_for_js_request,
+    prepare_nexus_for_js_request,
+)
+from cachito.workers.pkg_managers.npm import resolve_npm
+from cachito.workers.tasks.celery import app
+from cachito.workers.tasks.general import set_request_state
+
+
+__all__ = ["cleanup_npm_request", "fetch_npm_source"]
+log = logging.getLogger(__name__)
+
+
+@app.task
+def cleanup_npm_request(request_id):
+    """Clean up the Nexus npm content for the Cachito request."""
+    payload = {
+        "repository_name": get_js_proxy_repo_name(request_id),
+        "username": get_js_proxy_username(request_id),
+    }
+    nexus.execute_script("js_cleanup", payload)
+
+
+@app.task
+def fetch_npm_source(request_id, auto_detect=False):
+    """
+    Resolve and fetch npm dependencies for a given request.
+
+    :param int request_id: the Cachito request ID this is for
+    :param bool auto_detect: automatically detect if the archive uses npm
+    :raise CachitoError: if the task fails
+    """
+    validate_nexus_config()
```

Comment:
Let's do this at startup time instead.

Reply:
As discussed, there's no clean way to determine what queues the Celery worker is listening on since the configuration can be overridden with the `-Q` option.